### PR TITLE
load_earth_relief: Add the support of data source 'synbath'

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request adds the synbath data source to `load_earth_relief`

Addresses #1786

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
I haven't cached the new gridfiles yet; I'll add it later to `testing.py` assuming this addition is well-received.

**Comment by seisman** [comment]
I think you need to cache the `earth_synbat_` grids used in the tests.

File: pygmt/helpers/testing.py [hunk 1]
```diff
@@ -164,6 +164,7 @@ def download_test_data():
         "@earth_relief_10m_g",
         "@earth_relief_05m_p",
         "@earth_relief_05m_g",
+        "earth_synbath_01d_g.grd",
```
Comment (seisman) [inline pygmt/helpers/testing.py:167]:
```suggestion
        "@earth_synbath_01d_g",
```

File: pygmt/helpers/testing.py [hunk 2]
```diff
@@ -172,6 +173,8 @@ def download_test_data():
         "@N37W120.earth_relief_03s_g.nc",
         "@N00W090.earth_relief_03m_p.nc",
         "@N00E135.earth_relief_30s_g.nc",
+        # Earth synbath relief grids
+        "S15W105.earth_synbath_30s_p.nc",
```
Comment (seisman) [inline pygmt/helpers/testing.py:177]:
```suggestion
        "@S15W105.earth_synbath_30s_p.nc",
```

**Comment by willschlitzer** [comment]
I combined the SYNBATH 01d grid test with the IGPP test. Happy to split it back into two tests, but I think this is easier since that have the same shape and min/max values.

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
Since the PR title will be shown in the changelogs, what about changing the PR title to "load_earth_relief: Add the support of data source 'synbath'"?

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers for final reviews.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**